hudi-common/src/main/java/org/apache/hudi/stats/ValueType.java

```diff
@@ -244,77 +246,90 @@ public static ValueType fromParquetPrimitiveType(PrimitiveType primitiveType) {
     }
   }
 
-  public static ValueType fromSchema(Schema schema) {
-    switch (schema.getType()) {
+  /**
+   * Infers ValueType from HoodieSchema for type inference in column statistics.
+   * This method uses HoodieSchema for in-memory processing while maintaining
+   * compatibility with existing Avro-based serialization.
+   *
+   * @param schema the HoodieSchema to infer type from
+   * @return the corresponding ValueType
+   * @throws IllegalArgumentException if the schema type is not supported
+   * @since 1.2.0
+   */
+  public static ValueType fromSchema(HoodieSchema schema) {
```

Comment:
This should be easier to migrate once we have added the other types in this PR https://github.com/apache/hudi/pull/14312

Reply:
Let me rebase this PR onto your changes since they are merged.